src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -1261,12 +1263,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Return Nothing 'ConversionKind.NoConversion
                 End If
 
+                involvesNarrowingFromNumericConstant = involvesNarrowingFromNumericConstant Or elementConversion
```

Comment:
As we discussed offline, I will add a comment for the locals explaining the effect of optimization on the values that they contain.